### ISSUE TITLE
[rllib] Allow using PyTorch models with TF algorithms

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#3885]
### Describe the problem
It's possible to allow PyTorch models to be used with all RLlib's TF algorithms via a custom differentiable TF op: https://github.com/tensorflow/cleverhans/blob/master/cleverhans/utils_pytorch.py#L6

It's a little more work to get it working in practice, since you need to write a custom tf.train.Optimizer to apply the torch gradients, but it is worth investigating since it would allow torch to be used with all existing algs.

**Comment by ericl** [comment]
Won't fix.
